### ISSUE TITLE
Current Rasa for Botfront has critical bug in intent recognition

### ISSUE THREAD

**Issue by beevelop** [botfront/botfront#267]
The currently available release of Botfront has a severe bug in the intent recognition, that makes Rasa completely unusable. We experienced the bug, when upgrading from `v1.1.7-bf.4` to `v1.3.3-bf.6`. The intent recognition's accuracy of our test instances dropped below 30%. It turns out, that the confidence is somehow impacted and returns the same confidence value for a lot of intents.

I skimmed through the changelogs / PRs and issues of Rasa and finally found the related isssue: https://github.com/RasaHQ/rasa/issues/4573

The related PR with a bugfix has been merged: https://github.com/RasaHQ/rasa/pull/4581

Long story short: The current `v1.3.3-bf.6` can not be used. Upgrading Rasa to `v1.3.9` or manually applying the patch of the PR to the current `1.3.3` version should resolve the issue.

**Comment by znat** [comment]
@pheel dire 1.3.9 fix the CORS issue that made us roll back to 1.3.3?

**Comment by beevelop** [comment]
Not quite sure, which CORS issues appeared in your case, but there where a few fixes on that front too: https://github.com/RasaHQ/rasa/pull/4521 (https://github.com/RasaHQ/rasa/issues/4459)

**Comment by pheel** [comment]
Stay tuned for 1.3.9-bf.1!

**Comment by pheel** [comment]
https://github.com/botfront/rasa-for-botfront/releases/tag/v1.3.9-bf.1

**Comment by beevelop** [comment]
@pheel: Unfortunately I cannot confirm the issues being resolved with the changes of `v1.3.9-bf.1`. I tested the following setup / versions and the issue does still occur:
```
rasa-for-botfront:v1.3.9-bf.1
botfront:v0.16.4
botfront-api:v0.16.3
```

The plain Rasa (without BF patches) seems to work flawlessly in comparison.

**Comment by pheel** [comment]
The bugfix you mentioned earlier has been merged into the fork's codebase.

cf. https://github.com/botfront/rasa-for-botfront/commit/d19861a33dee58320c056473c01200eca0d10027 <= https://github.com/RasaHQ/rasa/commit/d19861a33dee58320c056473c01200eca0d10027

Most likely the problems you're encountering are due to something else. Can you share some more details?

**Comment by beevelop** [comment]
Yeah, I already checked the source files and the issue should be gone. Nonetheless, it still seems to appear in our case. We have several intents with the following naming scheme `faq.foobar.1`, `faq.foobar.2`, `faq.foobar.3`. The feature at least 2 example utterances that are quite distinctive. The training runs smoothly and the intent recognition for all other intent works flawlessly. But as soon as I put in an utterance from the `faq.foobar.*` intents, the confidence values are messed up: All intents starting with `faq.foobar.*` have the very same confidence. The F1 score drops below 40% and the model is unusable.

I tested the very same training data with the plain Rasa image. In `v1.3.8`, I was easily able to reproduce the behavior, whereas `v1.3.9` seems to resolve the issue properly. Unfortunately, I am not able to validate this with the Botfront-Rasa images, as I can't train directly from the CLI (an exception about the nlu_config,... is thrown).

**Comment by znat** [comment]
@beevelop can you paste the content of the `.botfront/botfront.yml` file?
If the Rasa image is the right one, we could try to repro the issue if you share a dataset/pipeline with us (hi_at_botfront.io)

**Comment by pheel** [comment]
@beevelop I'll look into this some more, but for the moment it seems like the bug only happens if your digit is directly preceded by a period. What is puzzling is that you don't have the same bug with vanilla 1.3.9...

**Comment by pheel** [comment]
At first face the problem doesn't seem to come from CountVectorsFeaturizer...

**Comment by beevelop** [comment]
Yep, I already tested the very same configuration with Vanilla Rasa: In `1.3.9` the issue does definitely not appear anymore. Whereas `1.3.8` seems to be affected by the behaviour described in the `CountVectorsFeaturizer` bug.

It seems to be the combination of a dot followed by a digit. If we replace the digits in the intent naming, the issue is gone.

**Comment by pheel** [comment]
Yep, sounds like something to report over at RasaHQ! Good find.

**Comment by martinavalogia** [comment]
Is this issue resolved??
